### PR TITLE
Add org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2

### PR DESCRIPTION
**Pull request by runiq**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/mikeoliphant/neural-amp-modeler-lv2/issues/33
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
  - I am not sure what to make of this requirement in the context of a Linux audio plugin such as this. I used the [template employed by other Linux audio plugins](https://github.com/flathub/org.freedesktop.LinuxAudio.BaseExtension). Is there something else I need to do here?
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*
  - The `nam-fixes.patch` file removes a temporary workaround that has been introduced for GCC versions < 9.3.0 (IIRC). The current FDO runtime ([22.08](https://github.com/flathub/org.freedesktop.LinuxAudio.BaseExtension/blob/ed3ef4db629eb34165eedb6240fbcfd986ca1333/org.freedesktop.LinuxAudio.BaseExtension.json#L4-L5) at the time of PR submission) uses [GCC 12.2](https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/blob/freedesktop-sdk-22.08.12.1/elements/bootstrap/include/gcc-source.yml?ref_type=tags#L2-4) which does not need that workaround anymore (in fact, leaving the workaround in place results in an error, aborting the build).

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

File: org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+branch: '22.08'
+build-extension: true
+runtime: org.freedesktop.LinuxAudio.BaseExtension
+runtime-version: stable
+sdk: org.freedesktop.Sdk//22.08
+build-options:
+  prefix: /app/extensions/Plugins/NeuralAmpModelerLV2
+modules:
+  - name: NeuralAmpModelerLV2
+    buildsystem: cmake
```
Comment (hfiguiere) [inline org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2.yml:11]:
```suggestion
    buildsystem: cmake-ninja
```
if it works

File: org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      - "install -Dm644 LICENCE.md ${FLATPAK_DEST}/share/licenses/NeuralAmpModelerLV2/LICENSE"
+      - "install -Dm644 deps/NeuralAmpModelerCore/LICENSE ${FLATPAK_DEST}/share/licenses/NeuralAmpModeler/LICENSE"
+      - "install -Dm644 --target-directory=${FLATPAK_DEST}/share/metainfo org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2.metainfo.xml"
+      - "appstream-compose --basename=org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2 --prefix=${FLATPAK_DEST} --origin=flatpak org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2"
+    cleanup:
+      - lib
+    sources:
+      - type: git
+        url: https://github.com/mikeoliphant/neural-amp-modeler-lv2
+        branch: main
```
Comment (hfiguiere) [inline org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2.yml:25]:
```suggestion
```

never a branch

File: org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="addon">
+  <id>org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2</id>
+  <name>Neural Amp Modeler LV2</name>
+  <summary>Neural Amp Modeler LV2 plugin implementation</summary>
```
Comment (hfiguiere) [inline org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2.metainfo.xml:5]:
```suggestion
  <summary>Neural Amp Modeler LV2 plugin</summary>
```

File: org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-3.0-only</project_license>
+  <extends>org.freedesktop.LinuxAudio.BaseExtension</extends>
+  <description>
+    <p>
+      Neural network emulator for guitar amplifiers.
+    </p>
+  </description>
```
Comment (hfiguiere) [inline org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2.metainfo.xml:13]: missing releases

Comment (hfiguiere) [reply org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2.metainfo.xml:13]: also missing `url`

File: org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2.metainfo.xml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="addon">
+  <id>org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2</id>
```
Comment (hfiguiere) [inline org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2.metainfo.xml:3]: LV2 shouldn't be part of the ID.

File: org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2.metainfo.xml [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+  <name>Neural Amp Modeler LV2</name>
```
Comment (hfiguiere) [inline org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2.metainfo.xml:4]:
```suggestion
  <name>Neural Amp Modeler</name>
```

**Comment by hfiguiere** [comment]
bot, build org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2

File: org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2.yml (hunk 2 above)
Comment (hfiguiere) [reply org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2.yml:25]: this is the cause of the build failure.

**Comment by hfiguiere** [comment]
you will also need to add a `flathub.json` with `    "skip-icons-check": true`. Like the other plugins.

File: org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2.yml (hunk 1 above)
Comment (runiq) [reply org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2.yml:11]: Is it still useful to use `cmake-ninja` here if the build system doesn't use ninja explicitly?

File: org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2.yml (hunk 2 above)
Comment (runiq) [reply org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2.yml:25]: Right, that was an oversight. Thanks for spotting it. <3

File: org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2.metainfo.xml (hunk 5 above)
Comment (runiq) [reply org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2.metainfo.xml:3]: There's a non-LV2 version of this plugin over at https://github.com/sdatkinson/NeuralAmpModelerPlugin. It's also from a different maintainer. My intention was to disambiguate the LV2 version from the non-LV2 version.

Given the above, should I still drop the LV2 bit? :thinking:

**Comment by runiq** [comment]
@hfiguiere Thanks a lot for the thorough review! <3 A quick question regarding the LV2 suffix: Should I drop it from the package name in general, i.e. should I use `org.freedesktop.LinuxAudio.Plugins.NeuralAmpModeler` in place of `...LV2` in the manifest file as well? (I guess that means I'll have to resubmit the PR with a different branch name too, right?)

File: org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2.metainfo.xml (hunk 5 above)
Comment (hfiguiere) [reply org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2.metainfo.xml]: no. the idea would be to add the format to the same package if anything.

**Comment by hfiguiere** [comment]
> (I guess that means I'll have to resubmit the PR with a different branch name too, right?)

no you wouldn't have to. Just change it.

File: org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2.metainfo.xml [hunk 7]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="addon">
+  <id>org.freedesktop.LinuxAudio.Plugins.NeuralAmpModeler</id>
```
Comment (hfiguiere) [inline org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2.metainfo.xml:3]: changing the ID should also be in the manifest and the files should be renamed too to match.

File: org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2.yml (hunk 1 above)
Comment (hfiguiere) [reply org.freedesktop.LinuxAudio.Plugins.NeuralAmpModelerLV2.yml]: cmake will be called with `-G Ninja` and the result is much faster build. However in some rare cases it doesn't work that's why it's an option.

**Comment by runiq** [comment]
Alright, I believe I've changed everything to remove the mention of LV2.

File: org.freedesktop.LinuxAudio.Plugins.NeuralAmpModeler.metainfo.xml [hunk 8]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="addon">
+  <id>org.freedesktop.LinuxAudio.Plugins.NeuralAmpModeler</id>
+  <name>Neural Amp Modeler</name>
+  <summary>Neural Amp Modeler plugin</summary>
```
Comment (hfiguiere) [inline org.freedesktop.LinuxAudio.Plugins.NeuralAmpModeler.metainfo.xml:5]:
```suggestion
  <summary>Neural Amp Modeler LV2 plugin</summary>
```

it actually belong in the summary.

File: org.freedesktop.LinuxAudio.Plugins.NeuralAmpModeler.metainfo.xml [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-3.0-only</project_license>
+  <extends>org.freedesktop.LinuxAudio.BaseExtension</extends>
+  <description>
+    <p>
+      Neural network emulator for guitar amplifiers.
```
Comment (hfiguiere) [inline org.freedesktop.LinuxAudio.Plugins.NeuralAmpModeler.metainfo.xml:11]:
```suggestion
      Neural network emulator for guitar amplifiers LV2 plugin.
```

File: org.freedesktop.LinuxAudio.Plugins.NeuralAmpModeler.yml [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+  - name: NeuralAmpModeler
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
+    post-install:
+      - "mv ${FLATPAK_DEST}/lib/lv2 ${FLATPAK_DEST}"
+      - "install -Dm644 LICENCE.md ${FLATPAK_DEST}/share/licenses/NeuralAmpModeler/LICENSE"
+      - "install -Dm644 deps/NeuralAmpModelerCore/LICENSE ${FLATPAK_DEST}/share/licenses/NeuralAmpModeler/LICENSE"
+      - "install -Dm644 --target-directory=${FLATPAK_DEST}/share/metainfo org.freedesktop.LinuxAudio.Plugins.NeuralAmpModeler.metainfo.xml"
+      - "appstream-compose --basename=org.freedesktop.LinuxAudio.Plugins.NeuralAmpModeler --prefix=${FLATPAK_DEST} --origin=flatpak org.freedesktop.LinuxAudio.Plugins.NeuralAmpModeler"
```
Comment (hfiguiere) [inline org.freedesktop.LinuxAudio.Plugins.NeuralAmpModeler.yml:19]:
```suggestion
      - "appstream-compose --basename=${FLATPAK_ID} --prefix=${FLATPAK_DEST} --origin=flatpak ${FLATPAK_ID}"
```

File: org.freedesktop.LinuxAudio.Plugins.NeuralAmpModeler.yml [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+  prefix: /app/extensions/Plugins/NeuralAmpModeler
+modules:
+  - name: NeuralAmpModeler
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
+    post-install:
+      - "mv ${FLATPAK_DEST}/lib/lv2 ${FLATPAK_DEST}"
+      - "install -Dm644 LICENCE.md ${FLATPAK_DEST}/share/licenses/NeuralAmpModeler/LICENSE"
+      - "install -Dm644 deps/NeuralAmpModelerCore/LICENSE ${FLATPAK_DEST}/share/licenses/NeuralAmpModeler/LICENSE"
```
Comment (hfiguiere) [inline org.freedesktop.LinuxAudio.Plugins.NeuralAmpModeler.yml:17]: the second line clobber the first one.

**Comment by hfiguiere** [comment]
bot, build org.freedesktop.LinuxAudio.Plugins.NeuralAmpModeler

File: org.freedesktop.LinuxAudio.Plugins.NeuralAmpModeler.yml (hunk 11 above)
Comment (runiq) [reply org.freedesktop.LinuxAudio.Plugins.NeuralAmpModeler.yml:17]: Good catch, thanks! I'll move the first license to the `NeuralAmpModelerLV2Plugin` subdirectory instead.

**Comment by runiq** [comment]
I'll try this too:

bot, build org.freedesktop.LinuxAudio.Plugins.NeuralAmpModeler

**Comment by runiq** [comment]
Heya, friendly ping :) Is there something else I need to do here, aside from testing? The plugin appears to work for me: I can run Carla, load the plugin, load an amp profile from tonehunt.org and jam away.

Should I maybe add this to the package description or something?

**Comment by hfiguiere** [comment]
patience is part of the process

**Comment by hfiguiere** [comment]
(build expired)

bot, build org.freedesktop.LinuxAudio.Plugins.NeuralAmpModeler

**Comment by hfiguiere** [comment]
Joy with submodules
```
fatal: Fetched in submodule path 'deps/NeuralAmpModelerCore', but it did not contain d72e8dc9bbf7331ad8998cc5fc02c8960cc13b93. Direct fetching of that commit failed.
```

**Comment by hfiguiere** [comment]
it fails locally too, now

**Comment by hfiguiere** [comment]
image

Submodule point to commit that doesn't exist on any branches. This is bound to fail.

**Comment by runiq** [comment]
> image
> 
> Submodule point to commit that doesn't exist on any branches. This is bound to fail.

Sorry, couldn't make time for this today, work came up. :( I'll have a look tomorrow, promise.

In any case, thanks again for the review!

File: org.freedesktop.LinuxAudio.Plugins.NeuralAmpModeler.yml [hunk 12]
```diff
@@ -0,0 +1,30 @@
+id: org.freedesktop.LinuxAudio.Plugins.NeuralAmpModeler
+branch: '22.08'
+build-extension: true
+runtime: org.freedesktop.LinuxAudio.BaseExtension
+runtime-version: stable
+sdk: org.freedesktop.Sdk//22.08
+build-options:
```
Comment (hfiguiere) [inline org.freedesktop.LinuxAudio.Plugins.NeuralAmpModeler.yml:7]:
```suggestion
appstream-compose: false
build-options:
```

File: org.freedesktop.LinuxAudio.Plugins.NeuralAmpModeler.yml [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+      - "install -Dm644 deps/NeuralAmpModelerCore/LICENSE ${FLATPAK_DEST}/share/licenses/NeuralAmpModeler/LICENSE"
+      - "install -Dm644 --target-directory=${FLATPAK_DEST}/share/metainfo org.freedesktop.LinuxAudio.Plugins.NeuralAmpModeler.metainfo.xml"
+      - "appstream-compose --basename=${FLATPAK_ID} --prefix=${FLATPAK_DEST} --origin=flatpak ${FLATPAK_ID}"
+    cleanup:
+      - lib
+    sources:
+      - type: git
+        url: https://github.com/mikeoliphant/neural-amp-modeler-lv2
+        tag: 0.1.0
+        commit: 0aad7c32b6806c826848b21bb1d2699eb79de29b
```
Comment (hfiguiere) [inline org.freedesktop.LinuxAudio.Plugins.NeuralAmpModeler.yml:26]:
```suggestion
        url: https://github.com/mikeoliphant/neural-amp-modeler-lv2
        commit: cde3df4e84b0a577d48b49de4aa451b1b3ef0171
```

Comment (hfiguiere) [reply org.freedesktop.LinuxAudio.Plugins.NeuralAmpModeler.yml:26]: this is the next commit that fix the submodules by using a more recent commit. At least this build.

File: org.freedesktop.LinuxAudio.Plugins.NeuralAmpModeler.yml (hunk 12 above)
Comment (hfiguiere) [reply org.freedesktop.LinuxAudio.Plugins.NeuralAmpModeler.yml:7]: not sure how it did build before but this is necessary.

**Comment by hfiguiere** [comment]
in principle it works here (in Ardour). Any preset you recommend? off the two I got I only get noise...

**Comment by runiq** [comment]
Argh, sorry! I deleted my Flathub fork and didn't realize it was still attached to this. I'll try and recreate.

**Pull request closed without merging**